bundles/org.eclipse.jface.text/src/org/eclipse/jface/text/contentassist/IContentAssistRequest.java

```diff
@@ -0,0 +1,55 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Dawid Pakuła and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Dawid Pakuła - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.jface.text.contentassist;
+
+import org.eclipse.jface.text.IDocument;
+import org.eclipse.jface.text.ITextViewer;
+
+/**
+ * Describes the state that the content assistant is in when completing proposals.
+ * <p>
+ * Clients may use this class.
+ * </p>
+ *
+ * @since 3.24
+ * @see IContentAssistProcessorExtension2
```

Comment:
Should have `@noimplements`

Reply:
This together with IContentAssistantExtension5 introduce more flexibility. For example JDT, PDT, DLTK, CDT build own ContentAssistInvocationContext to keep similar information (their processors have to be programmatically reqistered to assistant) . In this case will be possible to stop using extra ICompletionListener (see ContentAssistEvent).

Reply:
How an extensible interface would change the fact that computer that require custom context would need to plug it in the content assistant? And isn't the goal of this work exactly that the IContentAssistRequest to exactly cover all the informations that would allow a CompletionProcessor to pick the right choice, without need for extra context to be injected?

Reply:
I'm just looking for opportunity to via one patch,  make possible to always use built-in content assistant processor, especially with GenericTextEditor. In current state If you want introduce categorization (repeated invoke in popup) and/or add some extra information/filters in completion processor, you have to introduce ICompletionListener programmatically connected to concrete ContentAssistant. But for now I'll drop IContentAssistantExtension5 and start working on this, maybe for GenericEditorContentAssistant only

Reply:
Ok what now?